### ISSUE TITLE
BrowserWindow.fromWebContents returns undefined for windows created using window.open

### ISSUE THREAD

**Issue by pushkin-** [electron/electron#21608]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:** 5.0.7
* **Operating System:** Windows 10

### Expected Behavior
I expect that calling `BrowserWindow.fromWebContents(win.webContents)` returns `win`.

### Actual Behavior
The API returns `undefined`.

### To Reproduce
```
const { app, BrowserWindow } = require('electron')

async function createWindow() {
  var mainWindow = new BrowserWindow({
    webPreferences: {
      nativeWindowOpen: true
    }
  });
  await mainWindow.loadURL("https://google.com");
  mainWindow.webContents.openDevTools();
  console.log(BrowserWindow.fromWebContents(mainWindow.webContents)); // correctly returns mainWindow

  app.on("browser-window-created", (e, window) => {
    console.log(BrowserWindow.fromWebContents(window.webContents)); // returns undefined
    window.once("will-move", (e) => {
      console.log(BrowserWindow.fromWebContents(e.sender)); // extra test; also returns undefined
    });
  });
}

app.on('ready', createWindow)
```

Run `npm start` and open a window from devtools (e.g. `window.open("https://google.com")`). Notice that `fromWebContents` can't find the browser window for the new window.

**Comment by pushkin-** [comment]
I wonder if this will be fixed by [this PR](https://github.com/electron/electron/pull/21236)

**Comment by electron-triage** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron 6.x.y or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons. This issue will be closed 7 days from now if there is no response.

Thanks in advance! Your help is appreciated.

**Comment by pushkin-** [comment]
This is still an issue in Electron 8.

**Comment by pushkin-** [comment]
Still a problem in Electron 10.1.2

**Comment by pushkin-** [comment]
Still repros with Electron 12.0.1.

Though the call to `BrowserWindow.fromWebContents(e.sender)` now fails with:

![image](https://user-images.githubusercontent.com/6374473/111076404-2eddbf80-84ba-11eb-8386-287acc58b0d3.png)

**Comment by nornagon** [comment]
Here's a [bugbot](/electron/bugbot)-compatible gist: https://gist.github.com/decc0f7190ce719279b5f792711c7e6c

**Comment by nornagon** [comment]
/bugbot test https://gist.github.com/decc0f7190ce719279b5f792711c7e6c

**Comment by nornagon** [comment]
Fixed with #33257. The `will-move` handler in your example is incorrect as it passes a `BrowserWindow` object to `BrowserWindow.fromWebContents`, and the latter requires a `WebContents` as its argument.

Sorry it took so long to get to this! At least we fixed it eventually 😅
